### PR TITLE
Refacto info.fmt()

### PR DESCRIPTION
**Pull request by o2sh**
I refactored Info's display implementation by extracting some of the complexity into separate methods and removing `write_buf()` and `get_formatted_info_label()`.

### CODE REVIEW

File: src/onefetch/info.rs [hunk 1]
```diff
@@ -780,25 +646,110 @@ impl Info {
...
+        let pad = title.len() + 2;
+
+        for (i, author) in self.authors.iter().enumerate() {
+            if i == 0 {
+                author_field.push_str(&format!(
+                    "{}{} {} {}\n",
+                    author.2.to_string().color(self.color_set.info),
+                    "%".color(self.color_set.info),
+                    author.0.to_string().color(self.color_set.info),
+                    author.1.to_string().color(self.color_set.info),
```
Comment (spenserblack) [inline src/onefetch/info.rs:678]: Tuple destructuring might help make this more readable for future contributors.

File: src/onefetch/info.rs [hunk 2]
```diff
@@ -780,25 +646,110 @@ impl Info {
...
+        let tags_str = match self.number_of_tags {
+            0 => String::new(),
+            1 => String::from("1 tag"),
+            _ => format!("{} tags", self.number_of_tags),
+        };
+
+        if tags_str.is_empty() && branches_str.is_empty() {
+            String::new()
+        } else if branches_str.is_empty() || tags_str.is_empty() {
+            format!("({})", format!("{}{}", tags_str, branches_str))
```
Comment (spenserblack) [inline src/onefetch/info.rs:750]:
```suggestion
            format!("({}{})", tags_str, branches_str)
```

**Review by spenserblack — APPROVED** [review]
The code looks so much cleaner now :smile_cat: 

Left a couple of minor notes.

**Comment by o2sh** [comment]
> The code looks so much cleaner now :smile_cat:

Thanks :blush:

**Pull request merged**